### PR TITLE
Properly fetch `Gem#latest_spec_for` with multiple sources

### PR DESCRIPTION
**Pull request by kevlogan90**
# Description:
There are times when getting the latest spec for a gem is returning the wrong result. If multiple sources are used, and a gem exists in both, then [spec_fetcher](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/spec_fetcher.rb#L86) will return a spec for each source the gem is found in and [sort it by oldest to newest](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/spec_fetcher.rb#L136).

 Returning the last item from the list of specs resolves this issue.
______________

# Tasks:

- [x] Describe the problem / feature
- [x] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by bronzdoc** [comment]
Hey @kevlogan90 thanks for your contribution!

Could you add test for this change? thanks!

**Comment by kevlogan90** [comment]
Hey @bronzdoc I've added some tests. Please let me know if there are any changes that need to be made.

**Comment by bronzdoc** [comment]
@kevlogan90 i ran your tests without the code modifications you made and your tests still pass, so i'm not sure what is this fixing exactly.

**Comment by kevlogan90** [comment]
Sorry @bronzdoc , I see what you mean. I updated the test. It should fail for you now using the unmodified code.

File: test/rubygems/test_gem.rb [hunk 1]
```diff
@@ -764,6 +785,29 @@ def test_self_latest_version_for
...
+
+    spec_fetcher do |fetcher|
+      fetcher.spec 'a', 1
+      fetcher.spec 'a', 2
+      fetcher.spec 'a', '3.a'
+    end
+
+    version = Gem.latest_version_for 'a'
+
+    assert_equal Gem::Version.new(2), version
```
Comment (bronzdoc) [inline test/rubygems/test_gem.rb]: Same here, It seems that `a-2` is  only in one resource not both

File: test/rubygems/test_gem.rb [hunk 2]
```diff
@@ -740,6 +740,27 @@ def test_self_latest_spec_for
...
+      fetcher.spec 'a', 1.1
+    end
+
+    gems = spec_fetcher do |fetcher|
+      fetcher.spec 'a', 1
+      fetcher.spec 'a', '3.a'
+      fetcher.spec 'a', 2
+    end
+    spec = Gem.latest_spec_for 'a'
+    assert_equal gems['a-2'], spec
```
Comment (bronzdoc) [inline test/rubygems/test_gem.rb]: the problem happens when `multiple sources are used, and a gem exists in both`
but this test is not testing that, it seems that `a-2` is  only in one of the resources not both, i'm missing something?

Comment (kevlogan90) [reply test/rubygems/test_gem.rb]: Am I misunderstanding? The gem is `a`, with different versions `1`, `1.1`, `2`, `3.a`. It's possible for not all versions to be in all sources. In my case, which is likely an edge case, the gem was forked to an internal hosted rubygems source, and isn't updated regularly.

**Comment by deivid-rodriguez** [comment]
@kevlogan90 I'm sorry for the lack of feedback here? The PR looks good to me in principle, are you still interested? If so, please rebase it so we trigger the latest CI again? If not, please let us know so that we take it over.

**Comment by deivid-rodriguez** [comment]
No further updates, so I'll take over this PR. Thanks @kevlogan90!
